In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [ ]:
import os
os.chdir('/content/gdrive/MyDrive/kaggle')

In [ ]:
os.makedirs('./output')
os.makedirs('./output/train')
os.makedirs('./output/val')
os.makedirs('./output/test')
os.listdir('./output')

['train', 'val', 'test']

In [ ]:
import shutil
import random
import math
import numpy as np


In [ ]:
root_dir = '/content/gdrive/MyDrive/kaggle/chest_xray'
classes = ['CARDIOMEGALY', 'COVID', 'NORMAL', 'PNEUMONIA', 'PNEUMOTHORAX', 'TUBERCULOSIS']

In [ ]:
for clss in classes:
  print('------------' + clss + '-------------')
  dirtry = root_dir + '/' + clss
  files = os.listdir(dirtry)
  np.random.shuffle(files)
  base_outdir = '/content/gdrive/MyDrive/kaggle/output/'

  for folder in ['train', 'val', 'test']:
      target_dir = base_outdir + folder
      os.makedirs(target_dir + '/' + clss)
      target_class = target_dir + '/' + clss

      if folder == 'train':
          images_to_pass = files[: math.floor(0.8*len(files))]
          for img in images_to_pass:
              img = dirtry + '/' + img
              shutil.copy(img, target_class)
      elif folder == 'val':
          images_to_pass = files[math.floor(0.8*len(files)): math.floor(0.9*len(files))]
          for img in images_to_pass:
              img = dirtry + '/' + img
              shutil.copy(img, target_class)
      else:
          images_to_pass = files[math.floor(0.9*len(files)):]
          for img in images_to_pass:
              img = dirtry + '/' + img
              shutil.copy(img, target_class)
    



------------CARDIOMEGALY-------------
------------COVID-------------
------------NORMAL-------------
------------PNEUMONIA-------------
------------PNEUMOTHORAX-------------
------------TUBERCULOSIS-------------


In [ ]:
val_sum = 0
for flowr in os.listdir('./output/val'):
  lnk = './output/val/' + flowr
  val_sum += len(os.listdir(lnk))

train_sum = 0
for flowr in os.listdir('./output/train'):
  lnk = './output/train/' + flowr
  train_sum += len(os.listdir(lnk))

test_sum = 0
for flowr in os.listdir('./output/test'):
  lnk = './output/test/' + flowr
  test_sum += len(os.listdir(lnk))

print(val_sum)
print(train_sum)
print(test_sum)

1962
15706
1966


#**Dataset description:**

##Disease (class) split in data:

Total xray images: 19000

CARDIOMEGALY: 2784

COVID: 1525

NORMAL: 6842

PNEUMONIA: 4839

PNEUMOTHORAX: 2959

TUBERCULOSIS: 700

##Test, Train and Validation split:

Train: 15706

Test: 1966

Validation: 1962
